src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs

```diff
@@ -167,14 +167,32 @@ public void ApplyForwarders(HttpContext context)
             entryCount = Math.Max(forwardedHost.Length, entryCount);
         }
 
+        if (_options.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedPort))
+        {
+            checkPort = true;
+            forwardedPort = requestHeaders.GetCommaSeparatedValues(_options.ForwardedPortHeaderName);
+            if (_options.RequireHeaderSymmetry
+                && ((checkFor && forwardedFor!.Length != forwardedPrefix.Length)
```

Comment:
The intent here is to check that the For ones match the length of the Port ones you just got (same below).